client/CServerHandler.cpp

```diff
@@ -683,12 +683,24 @@ void CServerHandler::startCampaignScenario(std::shared_ptr<CampaignState> cs)
 		auto & epilogue = ourCampaign->scenario(*ourCampaign->lastScenario()).epilog;
 		auto finisher = [=]()
 		{
+			if(ourCampaign->campaignSet != "")
+			{
+				Settings entry = persistent.write["campaign"][ourCampaign->campaignSet][ourCampaign->getFilename()]["completed"];
```

Comment:
Perhaps simply
```suggestion
				Settings entry = persistent.write["completedCampaigns"][ourCampaign->getFilename()];
```
?
Don't expect more data in here other than completion status.

High scores AFAIK are not campaign/map based but global, across all maps, so should probably go into separate top-level config entry.